### ISSUE TITLE
Cannot restart Cobbler daemon if PXE menus exist

### ISSUE THREAD

**Issue by heroin-moose** [cobbler/cobbler#2692]
### Describe the bug

After creating a simple boot menu Cobbler cannot be restarted.

### Steps to reproduce

1. `cobbler distro add --name foo ...`
2. `cobbler profile add --name foo ...`
3. `cobbler menu add --name bar`
4. `cobbler profile edit --name foo --enable-menu 1 --menu bar`
5. `systemctl restart cobblerd`

### Expected behavior

It should be possible to restart Cobbler.

### Cobbler version

Latest Git master.

### Operating system

CentOS 8.

### Additional information

A quote from `journalctl -u cobblerd`:

```
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]: Traceback (most recent call last):
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/cobbler_collections/manager.py", line 189, in deserialize
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     serializer.deserialize(collection)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/serializer.py", line 128, in deserialize
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     storage_module.deserialize(collection, topological)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/modules/serializers/file.py", line 160, in deserialize
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     collection.from_list(datastruct)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/cobbler_collections/collection.py", line 201, in from_list
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     item = self.factory_produce(self.api, item_dict)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/cobbler_collections/profiles.py", line 45, in factory_produce
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     new_profile.from_dict(item_dict)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/items/profile.py", line 110, in from_dict
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     super().from_dict(dictionary)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/items/item.py", line 722, in from_dict
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     setattr(self, lowered_key, dictionary[key])
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/items/profile.py", line 690, in menu
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     raise CX("menu %s not found" % menu)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]: cobbler.cexceptions.CX: 'menu packer not found'
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]: The above exception was the direct cause of the following exception:
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]: Traceback (most recent call last):
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/bin/cobblerd", line 79, in main
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     api = cobbler_api.CobblerAPI(is_cobblerd=True)
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 107, in __init__
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     self.deserialize()
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 1616, in deserialize
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     return self._collection_mgr.deserialize()
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:   File "/usr/lib/python3.6/site-packages/cobbler/cobbler_collections/manager.py", line 192, in deserialize
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]:     % (collection.collection_type(), e)) from e
Jun 28 18:09:25 pxe.cobbler.lan cobblerd[16489]: cobbler.cexceptions.CX: "serializer: error loading collection profile: 'menu packer not found'. Check /etc/cobbler/modules.conf"
```

Also the JSON file with my menu looks kinda weird:

```
# jq < /var/lib/cobbler/collections//menus/packer.json
{
  "parent": "",
  "depth": 0,
  "children": [],
  "ctime": 1624806482.7926369,
  "mtime": 1624812434.7984204,
  "uid": "86a83ceb2e60437f85ea20b537c2b30f",
  "name": "packer",
  "comment": "",
  "kernel_options": {},
  "kernel_options_post": {},
  "autoinstall_meta": {},
  "fetchable_files": {},
  "boot_files": {},
  "template_files": {},
  "owners": [],
  "mgmt_classes": [],
  "mgmt_parameters": {},
  "is_subobject": false,
  "display_name": "Packer images",
  "ks_meta": {}
}
```

**Comment by SchoolGuy** [comment]
This is just an error in the order of loading items on startup. I will do it today.

**Comment by heroin-moose** [comment]
And is it normal that menu.json contains a lot of unused field like mgmt_classes or autoinstall_meta?

**Comment by SchoolGuy** [comment]
Yes because these are fields that are inhterited by `item.py`. That needs cleanup in one of the next versions indeed.

**Comment by nodeg** [comment]
This should be fixed with #2684. At least I can restart Cobbler successfully.

```bash
cat basic.sh                 
#!/bin/bash

# create fake kernel and initrd
sudo touch /root/vmlinuz
sudo touch /root/initrd

#cobbler menu add \
#        --name test \
#        --display-name "Test menu"

cobbler distro add \
        --name test-x86_64 \
        --arch x86_64 \
        --kernel /root/vmlinuz \
        --initrd /root/initrd \
        --boot-loaders 'grub ipxe' \

cobbler profile add \
        --name test-x86_64 \
        --distro test-x86_64 \
        --boot-loaders 'grub ipxe' \

cobbler system add \
        --name test \
        --profile test-x86_64 \
        --boot-loaders 'grub ipxe' \
```


```bash
$ cobbler menu add --name test

$ cobbler menu report
Name                           : test
Comment                        : 
Display Name                   : 
Parent Menu                    : 

$ cobbler profile list  
   test-x86_64

$ cobbler profile edit --name test-x86_64 --enable-menu 1 --menu test

$ cobbler profile report                                             
Name                           : test-x86_64
Automatic Installation Template : default.ks
Automatic Installation Metadata : <<inherit>>
TFTP Boot Files                : <<inherit>>
Boot loaders                   : ['grub', 'ipxe']
Comment                        : 
DHCP Tag                       : 
Distribution                   : test-x86_64
Enable iPXE?                   : False
Enable PXE Menu?               : True
Fetchable Files                : <<inherit>>
DHCP Filename Override         : <<inherit>>
Kernel Options                 : <<inherit>>
Kernel Options (Post Install)  : <<inherit>>
Parent boot menu               : test
Management Classes             : <<inherit>>
Management Parameters          : <<inherit>>
Name Servers                   : []
Name Servers Search Path       : []
Next Server (IPv4) Override    : 127.0.0.1
Next Server (IPv6) Override    : ::1
Owners                         : ['admin']
Parent Profile                 : 
Proxy                          : 
Red Hat Management Key         : <<inherit>>
Repos                          : []
Server Override                : 
Template Files                 : {}
Virt Auto Boot                 : True
Virt Bridge                    : xenbr0
Virt CPUs                      : 1
Virt Disk Driver Type          : raw
Virt File Size(GB)             : 0
Virt Path                      : 
Virt RAM (MB)                  : 512
Virt Type                      : xenpv
```

```bash
$ sudo systemctl restart cobblerd && systemctl status cobblerd
● cobblerd.service - Cobbler Helper Daemon
     Loaded: loaded (/etc/systemd/system/cobblerd.service; enabled; vendor preset: disabled)
     Active: active (running) since Mon 2021-07-12 11:36:22 CEST; 16ms ago
    Process: 13092 ExecStartPost=/usr/bin/touch /usr/share/cobbler/web/cobbler.wsgi (code=exited, status=1/FAILURE)
   Main PID: 13091 (cobblerd)
      Tasks: 1 (limit: 4615)
        CPU: 30ms
     CGroup: /system.slice/cobblerd.service
             └─13091 /usr/bin/python3 /usr/local/bin/cobblerd -F
```
